libs/crosstarget-utils/src/wasm/regex.rs

```diff
@@ -0,0 +1,39 @@
+use js_sys::RegExp as JSRegExp;
+
+use crate::common::{RegExpError, RegExpFlags};
+
+pub struct RegExp {
+    inner: JSRegExp,
+}
+
+impl RegExp {
+    pub fn new(pattern: &str, flags: Vec<RegExpFlags>) -> Result<Self, RegExpError> {
+        let mut flags_as_str: String = flags.into_iter().map(|flag| String::from(flag)).collect();
```

Comment:
```suggestion
        let mut flags_as_str: String = flags.into_iter().map(String::from).collect();
```
with this API, but I'd still prefer each flag to be converted to `&'static str` or `char` instead

Reply:
if we do refactor it to a bitmask though, then this whole logic would just be a method on `RegExpFlags` since there won't be an enum